### PR TITLE
Extensions of QRNN code.

### PR DESCRIPTION
**Pull request by simonpf**
This extends the current implementation of QRNN to support both Pytorch and Keras
as backends. The QRNN class has been refactored to separate general QRNN functionality
from the underlying neural network model. Some functionality has been added e.g. computing
the posterior mean and sampling from a fitted Gaussian.

This work is part of an upcoming study on cloud correction at 183 GHz.

### CODE REVIEW

File: doc/typhon.retrieval.qrnn.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+to the evaluation of QRNNs and the post processing of results such as computing
+the PSD or the posterior mean. For the rest it acts as a wrapper around its
+model attribute, which encapsules all network- and DL-framework-specific code.
+
+Backends
+--------
+
+Currently both `keras <https://keras.io/>`_ and `pytorch <https://pytorch.org/>`_
+are supported as backends for neural network. The QRNN implementation will
+automatically use the one the is available on your system. If both are available
```
Comment (olemke) [inline doc/typhon.retrieval.qrnn.rst]: Typo: the one *that* is

File: doc/typhon.retrieval.qrnn.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+.. automodule:: typhon.retrieval.qrnn.models.pytorch
+.. currentmodule:: typhon.retrieval.qrnn.models.pytorch
+.. autosummary::
+   :toctree: generated
+
+   FullyConnected
+   UNet
+
+.. automodule:: typhon.retrieval.qrnn.models.pytorch
+.. currentmodule:: typhon.retrieval.qrnn.models.pytorch
```
Comment (olemke) [inline doc/typhon.retrieval.qrnn.rst]: These should be `*.keras` instead of `*.pytorch` which is already documentated above.

**Review by olemke — CHANGES_REQUESTED** [review]
Just fix the two minor issues I mentioned and it's ok to merge from my side.

**Comment by simonpf** [comment]
Thank you for the suggestions! Fixed!

**Review by lkluft — APPROVED** [review]
Looks good 👍

**Review by olemke — APPROVED** [review]

**Pull request merged**